### PR TITLE
Refactor Checkbox and add readonly support

### PR DESCRIPTION
**Pull request by joshwooding**

### CODE REVIEW

**Review by alycrys — APPROVED** [review]

File: packages/core/src/checkbox/Checkbox.css [hunk 1]
```diff
@@ -1,17 +1,13 @@
...
-
 .saltCheckbox {
-  cursor: var(--salt-selectable-cursor-hover);
-  background: var(--saltCheckbox-background, none);
   display: flex;
-  gap: var(--saltCheckbox-gap, var(--salt-spacing-100));
+  gap: var(--salt-spacing-100);
   position: relative;
-  width: fit-content;
+  cursor: var(--salt-selectable-cursor-hover);
```
Comment (alycrys) [inline packages/core/src/checkbox/Checkbox.css:6]: shall we add `cursor: var(--salt-editable-cursor-readonly)`  when readonly?

Comment (alycrys) [reply packages/core/src/checkbox/Checkbox.css:6]: same for radio button

**Comment by alycrys** [comment]
Do you want to add a readonly checkbox in qa stories, as you did on radio button?

**Comment by joshwooding** [comment]
Yeah I need to add tests too

**Review by alycrys — APPROVED** [review]

**Review by Fercas123 — APPROVED** [review]

**Pull request merged**